File: ARCADE/Intro/Level 4_3: areSimilar?.cs
    using System;
     
    public class Test
    {
    	public static bool areSimilar(int[] A, int[] B){
                bool result = false;
                int index1 = -1, index2 = -1, index3 = -1;
                for (int i = 0; i < A.Length; i++){
                    if (A[i] != B[i]){
                        if (index1 == -1) index1 = i;
                        else{
                            if (index2 == -1) index2 = i;
                            else index3 = i;
                        }
                    }
                }
                if (index1 == -1) result = true;
                if (index1 != -1 && index2 != -1 && index3 == -1){
                    if (A[index1] == B[index2] && A[index2] == B[index1]) result = true;
                }
                return result;
            }
    	public static void Main()
    	{
    		int [] a = new int []{832, 998, 148, 570, 533, 561, 894, 147, 455, 279};
    		int [] b = new int []{832, 998, 148, 570, 533, 561, 455, 147, 894, 279};
    		Console.WriteLine(areSimilar(a, b));
    		int [] c = new int []{1, 2, 2};
    		int [] d = new int []{2, 1, 1};
    		Console.WriteLine(areSimilar(c,d));
    	}
    }

File: ARCADE/Intro/Level 4_3: areSimilar?.pas
program areSimilar;

Type
	ArrayInteger = Array of integer;
	
function areSimilar(a: ArrayInteger; b: ArrayInteger): boolean;
var index1, index2, index3, i : integer;
    rs : boolean;
begin
    rs := false;
    index1 := -1; index2 := -1; index3 := -1;
    for i := 0 to length(a) - 1 do
        if(a[i] <> b[i]) then
            if(index1 = -1) then index1 := i
            else
            begin
                if(index2 = -1) then index2 := i
                else index3 := i;
            end;
    
    if(index1 = -1) then rs := true;
    if((index1 <> -1) and (index2 <> -1) and (index3 = -1)) then
        if((a[index1] = b[index2]) and (a[index2] = b[index1])) then rs := true;
    areSimilar := rs;
end;

var a, b : ArrayInteger;
begin
	a := ArrayInteger.Create(1, 2, 3);
	b := ArrayInteger.Create(1, 2, 3);
	writeln(areSimilar(a,b));
	
	a := ArrayInteger.Create(1, 2, 3);
	b := ArrayInteger.Create(2, 1, 3);
	writeln(areSimilar(a,b));
	
	a := ArrayInteger.Create(1, 2, 2);
	b := ArrayInteger.Create(2, 1, 1);
	writeln(areSimilar(a,b));
end.
